net/src/packet/mod.rs

```diff
@@ -74,6 +74,11 @@ impl<Buf: PacketBufferMut> Packet<Buf> {
         &self.payload
     }
 
+    /// Add / Replace Ethernet header
+    pub fn set_eth(&mut self, eth: Eth) {
```

Comment:
technically unsafe

Reply:
How do you want to proceed? You want me to annotate all those methods as unsafe?